tests/org/mozilla/io/TestLocalMsgProtocol.java

```diff
@@ -353,26 +309,13 @@ public void test(TestHarness th) {
             // Scenario 1
             Thread serverCreateThread = new ThreadServerCreate();
             serverCreateThread.start();
-            synchronized (openLock) {
-                while (!serverCreated) {
-                    openLock.wait();
-                }
-            }
+            serverCreateThread.join();
             Thread serverAcceptAndOpenThread = new ThreadServerAcceptAndOpen();
             serverAcceptAndOpenThread.start();
-            synchronized (openLock) {
-                while (!serverAcceptAndOpenCalled) {
-                    openLock.wait();
-                }
-            }
-            Thread clientThread = new ThreadClient();
+            Thread clientThread = new ThreadClientCreate();
             clientThread.start();
-            serverCreateThread.join();
             serverAcceptAndOpenThread.join();
```

Comment:
Scenario 1 was:
1) Server creates a localmsg server
2) Server calls acceptAndOpen
3) Client opens a localmsg server

With this change, 3 could happen before 2.